**IMPORT** **DATA**

In [ ]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt


In [ ]:
#import the dataset
df1=pd.read_csv('/content/Merged file.csv')
df1

,Result_count,Page_URL,Site,Keyword,Product_name,Product_page,ASIN,Stars,Rating_count,Review_page,Current_price,Original_price,Shipping_price,Delivery,In_stock,Recent_purchase,Image,Current_progress
0,NaN,https://www.amazon.ca/s?k=Amazon+Device+Access...,Amazon.ca,Amazon Device Accessories,Amazon Basics Double Braided Nylon USB Type-C ...,NaN,B07D7TXTPP,4.6 out of 5 stars,NaN,NaN,$18.41,NaN,FREE delivery,"Sat, Jan 18",NaN,4.6 out of 5 stars,https://m.media-amazon.com/images/I/71Hw7458t8...,\n \n 36 results for...
1,NaN,https://www.amazon.ca/s?k=Amazon+Device+Access...,Amazon.ca,Amazon Device Accessories,BENFEI USB-C USB 3.0 to Gigabit Ethernet Adapt...,NaN,B0DC75R8FV,4.5 out of 5 stars,NaN,NaN,$19.99,NaN,FREE delivery,"Sat, Jan 18",NaN,200+ bought in past month,https://m.media-amazon.com/images/I/71RoH3G4mC...,\n \n 36 results for...
2,NaN,https://www.amazon.ca/s?k=Amazon+Device+Access...,Amazon.ca,Amazon Device Accessories,"Key Finders,ldcx 85dB Remote Finder Wireless I...",NaN,B085G96HZ2,4.2 out of 5 stars,NaN,NaN,$25.99,NaN,FREE delivery,"Sat, Jan 18",NaN,4.2 out of 5 stars,https://m.media-amazon.com/images/I/61fApZFPq0...,\n \n 36 results for...
3,NaN,https://www.amazon.ca/s?k=Amazon+Device+Access...,Amazon.ca,Amazon Device Accessories,MOSISO Compatible with MacBook Air 13 inch Cas...,NaN,B07KGMWX96,4.0 out of 5 stars,NaN,NaN,$17.99,NaN,FREE delivery,"Sat, Jan 18",NaN,300+ bought in past month,https://m.media-amazon.com/images/I/51EjWM0mjK...,\n \n 36 results for...
4,NaN,https://www.amazon.ca/s?k=Amazon+Device+Access...,Amazon.ca,Amazon Device Accessories,"Amazon Echo Pop (newest model), Full sound com...",NaN,B09WNK39JN,4.6 out of 5 stars,NaN,NaN,$54.99,NaN,FREE delivery,"Sat, Jan 18",NaN,5K+ bought in past month,https://m.media-amazon.com/images/I/61uz05Rgsu...,\n \n 36 results for...
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
11716,NaN,https://www.amazon.ca/s?k=Watches&page=7&crid=...,Amazon.ca,Watches,FANMIS,NaN,B00QAY4C4U,4.2 out of 5 stars,NaN,NaN,$29.99,NaN,NaN,NaN,Only 4 left in stock.,4.2 out of 5 stars,https://m.media-amazon.com/images/I/61ze63LJVF...,\n \n 289-306 of ove...
11717,NaN,https://www.amazon.ca/s?k=Watches&page=7&crid=...,Amazon.ca,Watches,VICTORIA HYDE,NaN,B072PQ2K49,3.7 out of 5 stars,NaN,NaN,$50.99,NaN,FREE delivery,"Sat, Jan 18",NaN,3.7 out of 5 stars,https://m.media-amazon.com/images/I/61Fc4CIoV9...,\n \n 289-306 of ove...
11718,NaN,https://www.amazon.ca/s?k=Watches&page=7&crid=...,Amazon.ca,Watches,Yapeach,NaN,B0CCTPWTLY,4.1 out of 5 stars,NaN,NaN,$28.99,NaN,NaN,NaN,NaN,4.1 out of 5 stars,https://m.media-amazon.com/images/I/61t1D2NrBy...,\n \n 289-306 of ove...
11719,NaN,https://www.amazon.ca/s?k=Watches&page=7&crid=...,Amazon.ca,Watches,OLEVS,NaN,B0C558V87S,4.1 out of 5 stars,NaN,NaN,$68.07,NaN,FREE delivery,"Sat, Jan 18",NaN,4.1 out of 5 stars,https://m.media-amazon.com/images/I/71PoRgZri6...,\n \n 289-306 of ove...


In [ ]:
df2=pd.read_csv('/content/Amazon Scraper Ecomm.csv')
df2

,Result_count,Page_URL,Site,Keyword,Product_name,Product_page,ASIN,Stars,Rating_count,Review_page,Current_price,Original_price,Shipping_price,Delivery,In_stock,Recent_purchase,Image,Current_progress
0,NaN,https://www.amazon.ca/s?k=Amazon+Device+Access...,Amazon.ca,Amazon Device Accessories,Amazon Basics Double Braided Nylon USB Type-C ...,NaN,B07D7TXTPP,4.6 out of 5 stars,NaN,NaN,$18.41,NaN,NaN,NaN,NaN,4.6 out of 5 stars,https://m.media-amazon.com/images/I/71Hw7458t8...,\n \n 35 results for...
1,NaN,https://www.amazon.ca/s?k=Amazon+Device+Access...,Amazon.ca,Amazon Device Accessories,"Wireless Charging Stand, GEEKERA 3 in 1 Wirele...",NaN,B087BY5HLB,4.3 out of 5 stars,NaN,NaN,$38.39,$47.99,FREE delivery,"Sat, Jan 18",NaN,2K+ bought in past month,https://m.media-amazon.com/images/I/61ljLiacM2...,\n \n 35 results for...
2,NaN,https://www.amazon.ca/s?k=Amazon+Device+Access...,Amazon.ca,Amazon Device Accessories,Apple MFi Certified 128GB Photo-Stick-for-iPho...,NaN,B07TMNRM9M,4.2 out of 5 stars,NaN,NaN,$43.15,$61.69,FREE delivery,"Sat, Jan 18",NaN,100+ bought in past month,https://m.media-amazon.com/images/I/51ZC3nvqKn...,\n \n 35 results for...
3,NaN,https://www.amazon.ca/s?k=Amazon+Device+Access...,Amazon.ca,Amazon Device Accessories,"(2024 Upgrade) Intoval Metal Charging Station,...",NaN,B0BM9LF96V,4.4 out of 5 stars,NaN,NaN,$47.99,NaN,FREE delivery,"Sat, Jan 18",NaN,50+ bought in past month,https://m.media-amazon.com/images/I/61DIenpDG6...,\n \n 35 results for...
4,NaN,https://www.amazon.ca/s?k=Amazon+Device+Access...,Amazon.ca,Amazon Device Accessories,"Amazon Echo Pop (newest model), Full sound com...",NaN,B09WNK39JN,4.6 out of 5 stars,NaN,NaN,$54.99,NaN,FREE delivery,"Sat, Jan 18",NaN,5K+ bought in past month,https://m.media-amazon.com/images/I/61uz05Rgsu...,\n \n 35 results for...
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
7655,NaN,https://www.amazon.ca/s?k=Outdoors&page=3&crid...,Amazon.ca,Outdoors,【2 Pack】 Solar Security Cameras Wireless Outdo...,NaN,B0DC2BS3LF,4.1 out of 5 stars,NaN,NaN,$199.99,NaN,FREE delivery,"Sat, Jan 18",NaN,4.1 out of 5 stars,https://m.media-amazon.com/images/I/71OrxHKagY...,\n \n 97-144 of over...
7656,NaN,https://www.amazon.ca/s?k=Outdoors&page=3&crid...,Amazon.ca,Outdoors,"Solar Security Camera Outdoor Wireless, GENBOL...",NaN,B0CGLKKRRM,4.2 out of 5 stars,NaN,NaN,$99.89,NaN,FREE delivery,"Sat, Jan 18",NaN,4.2 out of 5 stars,https://m.media-amazon.com/images/I/71zbYETefN...,\n \n 97-144 of over...
7657,NaN,https://www.amazon.ca/s?k=Outdoors&page=3&crid...,Amazon.ca,Outdoors,Flysocks Metal Detector for Adult - Profession...,NaN,B0D1X5QLNT,5.0 out of 5 stars,NaN,NaN,$119.99,NaN,FREE delivery,"Sat, Jan 18",NaN,5.0 out of 5 stars,https://m.media-amazon.com/images/I/71PkKvpiyD...,\n \n 97-144 of over...
7658,NaN,https://www.amazon.ca/s?k=Outdoors&page=3&crid...,Amazon.ca,Outdoors,Wanmingtek Colorful Christmas Tree Solar Wind ...,NaN,B0BGMBV2MW,4.6 out of 5 stars,NaN,NaN,$23.98,NaN,NaN,NaN,Only 6 left in stock.,4.6 out of 5 stars,https://m.media-amazon.com/images/I/61Zv7pzxZf...,\n \n 97-144 of over...


In [ ]:
df3=pd.read_csv('/content/Amazon Scraper Ecomm2.csv')
df3

,Result_count,Page_URL,Site,Keyword,Product_name,Product_page,ASIN,Stars,Rating_count,Review_page,Current_price,Original_price,Shipping_price,Delivery,In_stock,Recent_purchase,Image,Current_progress
0,NaN,https://www.amazon.ca/s?k=Personal+Computers&c...,Amazon.ca,Personal Computers,"Mini PC, Intel 12th Gen Alder Lake-N100(up to ...",NaN,B0DPP8QY79,5.0 out of 5 stars,NaN,NaN,$289.99,NaN,FREE delivery,"Sat, Jan 18",NaN,5.0 out of 5 stars,https://m.media-amazon.com/images/I/71miwxEUV1...,\n \n 1-48 of 180 re...
1,NaN,https://www.amazon.ca/s?k=Personal+Computers&c...,Amazon.ca,Personal Computers,"Mini PC, Beelink Mini S Intel 11th Gen 4-Cores...",NaN,B0B17LZ4GD,4.0 out of 5 stars,NaN,NaN,$159.00,$239.00,FREE delivery,"Sat, Jan 18",NaN,100+ bought in past month,https://m.media-amazon.com/images/I/51zfcIWibM...,\n \n 1-48 of 180 re...
2,NaN,https://www.amazon.ca/s?k=Personal+Computers&c...,Amazon.ca,Personal Computers,"Beelink Mini PC, Mini S12 Intel 12th Gen 4-Cor...",NaN,B0932CKPPY,4.2 out of 5 stars,NaN,NaN,$179.00,$249.00,FREE delivery,"Sat, Jan 18",NaN,50+ bought in past month,https://m.media-amazon.com/images/I/71RBiGNP7Q...,\n \n 1-48 of 180 re...
3,NaN,https://www.amazon.ca/s?k=Personal+Computers&c...,Amazon.ca,Personal Computers,Beelink MINI S12 Pro Mini PC Intel 12th Lake N...,NaN,B0DGQ5X7W5,4.2 out of 5 stars,NaN,NaN,$219.00,NaN,FREE delivery,"Sat, Jan 18",NaN,4.2 out of 5 stars,https://m.media-amazon.com/images/I/61URbz79rM...,\n \n 1-48 of 180 re...
4,NaN,https://www.amazon.ca/s?k=Personal+Computers&c...,Amazon.ca,Personal Computers,"Dell OptiPlex 3050 Desktop Computers PC, Intel...",NaN,B0C2CG426L,3.5 out of 5 stars,NaN,NaN,$247.99,NaN,$5.54 delivery,Jan 22 - 30,NaN,100+ bought in past month,https://m.media-amazon.com/images/I/61iT-gvYLJ...,\n \n 1-48 of 180 re...
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
3731,NaN,https://www.amazon.ca/s?k=Watches&page=7&crid=...,Amazon.ca,Watches,BY BENYAR,NaN,B08C2X57ZH,4.3 out of 5 stars,NaN,NaN,$49.99,NaN,FREE delivery,"Sat, Jan 18",NaN,4.3 out of 5 stars,https://m.media-amazon.com/images/I/71Mn7chOq9...,\n \n 289-306 of ove...
3732,NaN,https://www.amazon.ca/s?k=Watches&page=7&crid=...,Amazon.ca,Watches,LIGE,NaN,B0DCSN9FJK,4.3 out of 5 stars,NaN,NaN,$85.99,NaN,FREE delivery,"Sat, Jan 18",NaN,50+ bought in past month,https://m.media-amazon.com/images/I/81XPx013oP...,\n \n 289-306 of ove...
3733,NaN,https://www.amazon.ca/s?k=Watches&page=7&crid=...,Amazon.ca,Watches,V-LAFUYLIFE,NaN,B0DB57HBKS,3.4 out of 5 stars,NaN,NaN,$112.61,NaN,$66.88 delivery,"Tue, Jan 21",Only 6 left in stock.,3.4 out of 5 stars,https://m.media-amazon.com/images/I/61xgS4OEg0...,\n \n 289-306 of ove...
3734,NaN,https://www.amazon.ca/s?k=Watches&page=7&crid=...,Amazon.ca,Watches,LIGE,NaN,B0D794RNDY,4.1 out of 5 stars,NaN,NaN,$79.99,NaN,FREE delivery,"Sat, Jan 18",NaN,50+ bought in past month,https://m.media-amazon.com/images/I/81L6bIXwHt...,\n \n 289-306 of ove...


**COMBINE DATASETS**

In [ ]:
#concatenate the datasets to compare across them
combined_df=pd.concat([df1,df2,df3])
combined_df

,Result_count,Page_URL,Site,Keyword,Product_name,Product_page,ASIN,Stars,Rating_count,Review_page,Current_price,Original_price,Shipping_price,Delivery,In_stock,Recent_purchase,Image,Current_progress
0,NaN,https://www.amazon.ca/s?k=Amazon+Device+Access...,Amazon.ca,Amazon Device Accessories,Amazon Basics Double Braided Nylon USB Type-C ...,NaN,B07D7TXTPP,4.6 out of 5 stars,NaN,NaN,$18.41,NaN,FREE delivery,"Sat, Jan 18",NaN,4.6 out of 5 stars,https://m.media-amazon.com/images/I/71Hw7458t8...,\n \n 36 results for...
1,NaN,https://www.amazon.ca/s?k=Amazon+Device+Access...,Amazon.ca,Amazon Device Accessories,BENFEI USB-C USB 3.0 to Gigabit Ethernet Adapt...,NaN,B0DC75R8FV,4.5 out of 5 stars,NaN,NaN,$19.99,NaN,FREE delivery,"Sat, Jan 18",NaN,200+ bought in past month,https://m.media-amazon.com/images/I/71RoH3G4mC...,\n \n 36 results for...
2,NaN,https://www.amazon.ca/s?k=Amazon+Device+Access...,Amazon.ca,Amazon Device Accessories,"Key Finders,ldcx 85dB Remote Finder Wireless I...",NaN,B085G96HZ2,4.2 out of 5 stars,NaN,NaN,$25.99,NaN,FREE delivery,"Sat, Jan 18",NaN,4.2 out of 5 stars,https://m.media-amazon.com/images/I/61fApZFPq0...,\n \n 36 results for...
3,NaN,https://www.amazon.ca/s?k=Amazon+Device+Access...,Amazon.ca,Amazon Device Accessories,MOSISO Compatible with MacBook Air 13 inch Cas...,NaN,B07KGMWX96,4.0 out of 5 stars,NaN,NaN,$17.99,NaN,FREE delivery,"Sat, Jan 18",NaN,300+ bought in past month,https://m.media-amazon.com/images/I/51EjWM0mjK...,\n \n 36 results for...
4,NaN,https://www.amazon.ca/s?k=Amazon+Device+Access...,Amazon.ca,Amazon Device Accessories,"Amazon Echo Pop (newest model), Full sound com...",NaN,B09WNK39JN,4.6 out of 5 stars,NaN,NaN,$54.99,NaN,FREE delivery,"Sat, Jan 18",NaN,5K+ bought in past month,https://m.media-amazon.com/images/I/61uz05Rgsu...,\n \n 36 results for...
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
3731,NaN,https://www.amazon.ca/s?k=Watches&page=7&crid=...,Amazon.ca,Watches,BY BENYAR,NaN,B08C2X57ZH,4.3 out of 5 stars,NaN,NaN,$49.99,NaN,FREE delivery,"Sat, Jan 18",NaN,4.3 out of 5 stars,https://m.media-amazon.com/images/I/71Mn7chOq9...,\n \n 289-306 of ove...
3732,NaN,https://www.amazon.ca/s?k=Watches&page=7&crid=...,Amazon.ca,Watches,LIGE,NaN,B0DCSN9FJK,4.3 out of 5 stars,NaN,NaN,$85.99,NaN,FREE delivery,"Sat, Jan 18",NaN,50+ bought in past month,https://m.media-amazon.com/images/I/81XPx013oP...,\n \n 289-306 of ove...
3733,NaN,https://www.amazon.ca/s?k=Watches&page=7&crid=...,Amazon.ca,Watches,V-LAFUYLIFE,NaN,B0DB57HBKS,3.4 out of 5 stars,NaN,NaN,$112.61,NaN,$66.88 delivery,"Tue, Jan 21",Only 6 left in stock.,3.4 out of 5 stars,https://m.media-amazon.com/images/I/61xgS4OEg0...,\n \n 289-306 of ove...
3734,NaN,https://www.amazon.ca/s?k=Watches&page=7&crid=...,Amazon.ca,Watches,LIGE,NaN,B0D794RNDY,4.1 out of 5 stars,NaN,NaN,$79.99,NaN,FREE delivery,"Sat, Jan 18",NaN,50+ bought in past month,https://m.media-amazon.com/images/I/81L6bIXwHt...,\n \n 289-306 of ove...


**IDENTICAL DATA **

In [ ]:
#check identical rows across all dataset
ident_rows = combined_df[combined_df.duplicated(keep=False)]
ident_rows

,Result_count,Page_URL,Site,Keyword,Product_name,Product_page,ASIN,Stars,Rating_count,Review_page,Current_price,Original_price,Shipping_price,Delivery,In_stock,Recent_purchase,Image,Current_progress
58,NaN,https://www.amazon.ca/s?k=Amazon+Kindle&crid=2...,Amazon.ca,Amazon Kindle,Amazon,NaN,B09BSQ8PRD,4.3 out of 5 stars,NaN,NaN,$509.99,NaN,FREE delivery,"Tue, Jan 21",NaN,500+ bought in past month,https://m.media-amazon.com/images/I/71W31zjQpY...,\n \n 1-48 of over 2...
59,NaN,https://www.amazon.ca/s?k=Amazon+Kindle&crid=2...,Amazon.ca,Amazon Kindle,New Amazon Kindle Scribe (64GB) - Redesigned d...,NaN,B0CZ9TBL1H,3.2 out of 5 stars,NaN,NaN,$569.99,NaN,FREE delivery,"Sat, Jan 18",NaN,100+ bought in past month,https://m.media-amazon.com/images/I/51dg1yuvH8...,\n \n 1-48 of over 2...
61,NaN,https://www.amazon.ca/s?k=Amazon+Kindle&crid=2...,Amazon.ca,Amazon Kindle,Kindle Essentials Bundle including Kindle (202...,NaN,B0DB2R95S1,4.5 out of 5 stars,NaN,NaN,$192.97,$212.97,NaN,NaN,NaN,4.5 out of 5 stars,https://m.media-amazon.com/images/I/81FI+l8h7k...,\n \n 1-48 of over 2...
72,NaN,https://www.amazon.ca/s?k=Amazon+Kindle&crid=2...,Amazon.ca,Amazon Kindle,Amazon,NaN,B0C2XT5ZPB,4.2 out of 5 stars,NaN,NaN,$144.99,$189.99,FREE delivery,"Sat, Jan 18",NaN,100+ bought in past month,https://m.media-amazon.com/images/I/51-6uBmjPq...,\n \n 1-48 of over 2...
73,NaN,https://www.amazon.ca/s?k=Amazon+Kindle&crid=2...,Amazon.ca,Amazon Kindle,Amazon,NaN,B09BSQ8PRD,4.3 out of 5 stars,NaN,NaN,$509.99,NaN,FREE delivery,"Tue, Jan 21",NaN,500+ bought in past month,https://m.media-amazon.com/images/I/71W31zjQpY...,\n \n 1-48 of over 2...
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
3418,NaN,https://www.amazon.ca/s?k=Watches&page=2&xpid=...,Amazon.ca,Watches,GOLDEN HOUR,NaN,B07G4Z8MHC,4.1 out of 5 stars,NaN,NaN,$44.99,NaN,FREE delivery,"Sat, Jan 18",NaN,4.1 out of 5 stars,https://m.media-amazon.com/images/I/71AjORbPfL...,\n \n 49-96 of over ...
3443,NaN,https://www.amazon.ca/s?k=Watches&page=3&crid=...,Amazon.ca,Watches,GOLDEN HOUR,NaN,B078S38KXZ,4.1 out of 5 stars,NaN,NaN,$44.99,NaN,FREE delivery,"Sat, Jan 18",NaN,4.1 out of 5 stars,https://m.media-amazon.com/images/I/715Gdgija8...,\n \n 97-144 of over...
3483,NaN,https://www.amazon.ca/s?k=Watches&page=3&crid=...,Amazon.ca,Watches,GOLDEN HOUR,NaN,B078S38KXZ,4.1 out of 5 stars,NaN,NaN,$44.99,NaN,FREE delivery,"Sat, Jan 18",NaN,4.1 out of 5 stars,https://m.media-amazon.com/images/I/715Gdgija8...,\n \n 97-144 of over...
3507,NaN,https://www.amazon.ca/s?k=Watches&page=4&crid=...,Amazon.ca,Watches,Suitisbest,NaN,B0BZRVPT61,4.4 out of 5 stars,NaN,NaN,$20.99,NaN,FREE delivery,"Sat, Jan 18",NaN,500+ bought in past month,https://m.media-amazon.com/images/I/81ZPbKCfOX...,\n \n 145-192 of ove...


In [ ]:
#number of identical rows
num_ident_rows = ident_rows.shape[0]
num_ident_rows

370

Out of 23117 rows across the three datasets, only 370 rows are identical. This indicates that are relatively small subset of the data appears consistently across all three datasets, while majority of the row differs.